server/src/internalClusterTest/java/org/elasticsearch/index/mapper/IgnoreFieldIT.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.index.query.ExistsQueryBuilder;
+import org.elasticsearch.index.query.IdsQueryBuilder;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.aggregations.bucket.terms.StringTerms;
+import org.elasticsearch.search.aggregations.metrics.InternalAvg;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.junit.Before;
+
+import static org.elasticsearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.avg;
+import static org.elasticsearch.search.aggregations.AggregationBuilders.terms;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.hasSize;
+
+@SuppressWarnings("resource")
+public class IgnoreFieldIT extends ESIntegTestCase {
```

Comment:
does this need to be an integ test case? Could it be a single node test case instead? Even better, there are probably parts that could become unit test and not need to spin up a cluster with multiple nodes? Can you check the existing coverage for _ignored in the existing yaml tests? Shall we add a basic yaml test that aggregates on it? We also need to add a bw compatibility test that verifies the behaviour against a mixed cluster.

Reply:
I think I need to add a few tests where I run at least a handful of aggregations on it...if I try to imagine possible use case scenarios I imagine top[Hits/Metrics] aggs maybe nested in daily/weekly/monthly histograms...date range, histogram and cardinality at least.